src/main/kotlin/io/snyk/plugin/ui/jcef/ApplyFixHandler.kt

```diff
@@ -0,0 +1,215 @@
+package io.snyk.plugin.ui.jcef
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.ui.jcef.JBCefBrowserBase
+import com.intellij.ui.jcef.JBCefJSQuery
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.cef.browser.CefBrowser
+import org.cef.browser.CefFrame
+import org.cef.handler.CefLoadHandlerAdapter
+
+data class DiffPatch(
+    val originalFile: String,
+    val fixedFile: String,
+    val hunks: List<Hunk>
+)
+
+data class Hunk(
+    val startLineOriginal: Int,
+    val numLinesOriginal: Int,
+    val startLineFixed: Int,
+    val numLinesFixed: Int,
+    val changes: List<Change>
+)
+
+sealed class Change {
+    data class Addition(val line: String) : Change()
+    data class Deletion(val line: String) : Change()
+    data class Context(val line: String) : Change()  // Unchanged line for context
+}
+
+class ApplyFixHandler(private val project: Project) {
+
+    fun generateApplyFixCommand(jbCefBrowser: JBCefBrowserBase): CefLoadHandlerAdapter {
+        val applyFixQuery = JBCefJSQuery.create(jbCefBrowser)
+
+        applyFixQuery.addHandler { value ->
+            val params = value.split(":")
+            val filePath = params[0]  // Path to the file that needs to be patched
+            val patch = params[1]      // The patch we received from LS
+
+            println("[applyFixHandler] Received request to apply fix on file: $filePath")
+            println("[applyFixHandler] Patch to apply: $patch")
+
+
+            // Avoid blocking the UI thread
+            CoroutineScope(Dispatchers.IO).launch {
+                try {
+                    val success = applyPatchAndSave(project, filePath, patch)
+
+                    val script = """
+                        window.receiveApplyFixResponse($success);
+                    """.trimIndent()
+
+                    withContext(Dispatchers.Main) {
+                        jbCefBrowser.cefBrowser.executeJavaScript(script, jbCefBrowser.cefBrowser.url, 0)
+                    }
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                }
+            }
+            return@addHandler JBCefJSQuery.Response("success")
+        }
```

Comment:
You can also use `runAsync`. I'm confident that `kotlinx` is available in all IDEA IDEs - but we need to test this in a non Kotlin IDE, e.g. Goland/Pycharm.